third_party/git-cmake-format/CMakeLists.txt

```diff
@@ -3,10 +3,14 @@ include(FetchContent)
 FetchContent_Declare(
     git_cmake_format
     GIT_REPOSITORY https://github.com/ginkgo-project/git-cmake-format.git
-    GIT_TAG        be9554a5d71030dcc23f1eb355c41ab9d876e776
+    GIT_TAG        069338e8afd47e6a27e661940cd36e940253f81c
 )
 FetchContent_GetProperties(git_cmake_format)
 if(NOT git_cmake_format_POPULATED)
     FetchContent_Populate(git_cmake_format)
+
+    if(NOT DEFINED GCF_FORCE_OVERWRITE)
+        set(GCF_FORCE_OVERWRITE ON CACHE INTERNAL "If true, always overwrite pre-commit hook and script")
+    endif()
```

Comment:
I think it needs to be before calling `FetchContent_Populate` or `FetchContent_GetProperties`.
Using the defined might be confusing.
`cmake -DGCF_FORCE_OVERWRITE=ON ..` it still turn off GCF_FORCE_OVERWRITE.
I am not sure whether FetchContent can realize the config change in the second time or not

Reply:
As far as I understand it, FetchContent does not execute any cmake code of the fetched content. That happens only in the `add_subdirectory` call, so I think it should be fine to set the variable here. I'm not sure what you mean in your `cmake -DGCF_FORCE_OVERWRITE=ON ..` part (assuming you also set `GINKGO_DEVEL_TOOLS`). One problem might be the `INTERNAL` option, which hides `GCF_FORCE_OVERWRITE` at least when using ccmake. But at least for me, it works as expected, i.e. setting it to `ON` if `-DGCF_FORCE_OVERWRITE` is not used and using that value otherwise.

Reply:
does cmake allow -DGCF_FORCE_OVERWRITE directly?
my cmake complain it `VAR:type=value`
I mean `cmake -DGCF_FORCE_OVERWRITE=OFF ..`  and `cmake -DGCF_FORCE_OVERWRITE=OFF ..` give the same behavior

Reply:
Which cmake version are you using, I'm using 3.17.0 without complaints? Also does changing `INTERNAL` to `BOOL` change the behavior?

Reply:
cmake 3.13.

I rerun it again. it does not work as expected due to my typo :man_facepalming:
and use -DGCF_FORCE_OVERWRITE=ON/OFF will overwrite the internal one, and it will change the setting in git-cmake-format.
Is empty for OFF or ON in cmake?

Reply:
Empty would correspond to OFF, if you test it in an if condition, i.e. in `if(<var>)` it would evaluate to false if `<var>` is empty or not defined.

Reply:
but it will depends on how you detect the VARIABLE.
anyway, it is the cmake usage and git-cmake-format claims it should be ON/OFF